enterprise/models.py

```diff
@@ -272,6 +272,37 @@ def user_email(self):
             return self.user.email
         return None
 
+    @property
+    def entitlements(self):
```

Comment:
> When you say "cannot" here, you mean "cannot currently due to how we implemented course-level consent", rather than "cannot in principle", right?

Not what I mean at all. I mean that even _asking_ for consent for a specific course isn't meaningful unless the user has tried to interact with that course. So if the question this API is trying to answer is "which coupons is this user allowed to interact with/attempt to use?", then we ought not to gate on the fact that course-specific consent is required.

On the other hand, if the intended question is "which coupons can be immediately applied without further user intervention?", then yes, we ought to gate on consent.

All of this is the entire reason I asked about the specific use-case for this API. Obviously, the two main ones have very different requirements.

@saleem-latif, perhaps we can fit a greater number of use cases with this API by returning something like the below?

```json
{
    "entitlements": [
        {
            "requires_consent": true,
            "entitlement_id": 87
        },
        {
            "requires_consent": false,
            "entitlement_id": 32
        }
    ]
}
```